src/WebJobs.Script.WebHost/Middleware/AppServiceHeaderFixupMiddleware.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.Extensions;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    public class AppServiceHeaderFixupMiddleware
+    {
+        private const string DisguisedHostHeader = "DISGUISED-HOST";
+        private const string HostHeader = "HOST";
+        private const string ForwardedProtocolHeader = "X-Forwarded-Proto";
+        private readonly RequestDelegate _next;
+
+        public AppServiceHeaderFixupMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext httpContext)
+        {
+            StringValues value;
+            if (httpContext.Request.Headers.TryGetValue(DisguisedHostHeader, out value))
```

Comment:
nit: you can also inline this (i.e. `httpContext.Request.Headers.TryGetValue(DisguisedHostHeader, out StringValues value)`

Reply:
yeah, I was gonna use that until I was surprised by the scope of that `value` ref. I thought it would be in the if statement, but it actually has the function scope, so it'll have to be 

```cs
if (httpContext.Request.Headers.TryGetValue(DisguisedHostHeader, out StringValues value))
{
}

if (httpContext.Request.Headers.TryGetValue(ForwardedProtocolHeader, out value))
{
}
```

which I thought looked odd. But I'll update it

Reply:
I'm adding an integration test now